src/cpp/rtps/history/TopicPayloadPool.cpp

```diff
@@ -278,7 +278,7 @@ bool TopicPayloadPool::shrink (
 }
 
 std::shared_ptr<ITopicPayloadPool> TopicPayloadPool::get(
-        PoolConfig config)
+        const PoolConfig& config)
```

Comment:
I'm thinking we start to overuse `PoolConfig` as the input parameter (here, and as a consequence of dependencies, also in the registry). We are only using the `memory_policy` and `payload_initial_size`. Having the `initial_size` and `maximum_size` hanging around looks distracting.

Just a crazy idea: Should we have a `history_config` with all 4 values and a `pool_config` with just the first two?

Reply:
I've refactored `PoolConfig` to have a base `BasicPoolConfig`